cql3/expr/expression.hh

```diff
@@ -208,6 +208,11 @@ struct bind_variable {
     // FIXME: infer shape from expression rather than from grammar
     shape_type shape;
     int32_t bind_index;
+
+    // Type of the bound value.
+    // Before preparing can be nullptr.
+    // After preparing always holds a valid type.
+    data_type value_type;
```

Comment:
This can be merged with the patch that assigns it.

Reply:
done